command/src/main/java/pw/chew/jdachewtils/command/OptionHelper.java

```diff
@@ -30,173 +30,199 @@
-        List<OptionMapping> options = event.getOptionsByName(option);
+    public static Member optMember(@NotNull SlashCommandEvent event, @NotNull String key, @Nullable Member defaultValue) {
+        if (!event.isFromGuild())
+            return defaultValue; // Non-guild commands do not have a member.
+        
+        OptionMapping option = event.getOption(key);
 
-        return options.isEmpty() ? defaultValue : options.get(0).getAsMember();
+        return option == null ? defaultValue : option.getAsMember();
     }
 
     /**
-     * Guarantees a IMentionable option value by providing a default value.
+     * Gets the provided Option Key as a IMentionable value, or returns the default one if the option cannot be found.
      *
      * @param event        The slash command event to get options from
-     * @param option       The option we want
-     * @param defaultValue The fallback option in case of the absence of the option value
+     * @param key          The option we want
+     * @param defaultValue Nullable default value used in the absence of the option value
      * @return The provided option, or the default value if the option is not present
      */
     @Nullable
     @Contract("_, _, !null -> !null")
-    public static IMentionable optMentionable(@NotNull SlashCommandEvent event, @NotNull String option, @Nullable IMentionable defaultValue) {
-        List<OptionMapping> options = event.getOptionsByName(option);
+    public static IMentionable optMentionable(@NotNull SlashCommandEvent event, @NotNull String key, @Nullable IMentionable defaultValue) {
+        OptionMapping option = event.getOption(key);
 
-        return options.isEmpty() ? defaultValue : options.get(0).getAsMentionable();
+        return option == null ? defaultValue : option.getAsMentionable();
     }
 
     /**
-     * Guarantees a Role option value by providing a default value.
+     * Gets the provided Option Key as a Role value, or returns the default one if the option cannot be found.
      *
      * @param event        The slash command event to get options from
-     * @param option       The option we want
-     * @param defaultValue The fallback option in case of the absence of the option value
+     * @param key          The option we want
+     * @param defaultValue Nullable default value used in the absence of the option value
      * @return The provided option, or the default value if the option is not present
      */
     @Nullable
     @Contract("_, _, !null -> !null")
-    public static Role optRole(@NotNull SlashCommandEvent event, @NotNull String option, @Nullable Role defaultValue) {
-        List<OptionMapping> options = event.getOptionsByName(option);
+    public static Role optRole(@NotNull SlashCommandEvent event, @NotNull String key, @Nullable Role defaultValue) {
+        OptionMapping option = event.getOption(key);
```

Comment:
Fair. Will change that once I'm back home today

Reply:
Any updates on this? 😄

Reply:
Sorry about that. I honestly just forgot...

But it should now be added. And while we're here. I was thinking... would it be a good idea to have some `optX` methods without a defaultValue to define? Those would then just return a specific value (`null` for everything nullable or `0` for numbers and such) in such a case.

Could simplify end-user code since they don't have to provide null every single time...

Reply:
I mean, you should always provide a default value (in theory). But I don't see why not. The optX(event, name) methods should just call the respective optX(event, name, `null`) to save on code, though.